examples/load_from_cloud.html

```diff
@@ -0,0 +1,487 @@
+      bufView[i] = str.charCodeAt(i);
+    }
+    return buf;
+  }
+
+  function importPrivateKey(pem) {
+    // fetch the part of the PEM string between header and footer
+    const pemHeader = "-----BEGIN PRIVATE KEY-----";
+    const pemFooter = "-----END PRIVATE KEY-----";
+    const pemContents = pem.substring(pemHeader.length, pem.length - pemFooter.length);
+    // base64 decode the string to get the binary data
+    const binaryDerString = window.atob(pemContents);
+    // convert from a binary string to an ArrayBuffer
+    const binaryDer = str2ab(binaryDerString);
+    const keyUsages = ['sign'];
+
+    return window.crypto.subtle.importKey(
+      "pkcs8",
+      binaryDer,
+      {
+        name: "RSASSA-PKCS1-v1_5",
+        hash: "SHA-256",
+      },
+      true,
+      keyUsages
+    );
+  }
+
+  async function rsaSha256(key, data) {
+    const algorithm = "RSASSA-PKCS1-v1_5";
+    const keyObj = await importPrivateKey(key);
+    const enc = new TextEncoder(); // always utf-8
+    const encodedBufferArray = enc.encode(data);
+    return crypto.subtle.sign(algorithm, keyObj, encodedBufferArray);
+  }
+
+  function sha256Encode(arrBuff) {
+  return crypto.subtle.digest("SHA-256", arrBuff);
+  }
+
+  // code snippet taken from: https://stackoverflow.com/questions/40031688/javascript-arraybuffer-to-hex
+  // licensed under a MIT license
+  function buf2hex(buffer) { // buffer is an ArrayBuffer
+    return Array.prototype.map.call(new Uint8Array(buffer), x => ('00' + x.toString(16)).slice(-2)).join('');
+  }
+
+  // sha-256 hash and hex encode the canonical request
+  async function encodeCanonical(request) {
+    const enc = new TextEncoder(); // always utf-8
+    const encodedBufferArray = enc.encode(request);
+    const encodedCanonicalArrBuff = await sha256Encode(encodedBufferArray);
+    const encodedCanonicalHex = buf2hex(encodedCanonicalArrBuff);
+
+    return encodedCanonicalHex;
+  }
+
+  async function getGCPSignedUrl(url, headers, privateKeyFile, bucket) {
+
+    // extract the private key from the given file
+    const keyHeader = '"private_key": "';
+    const keyFooter = '\\n",';
+    const keyStart = privateKeyFile.indexOf(keyHeader) + keyHeader.length;
+    const keyEnd = privateKeyFile.indexOf(keyFooter);
+    const privateKey = privateKeyFile.substring(keyStart, keyEnd).replaceAll("\\n", "\n");
+
+    // extract the client email from the given file
+    const emailHeader = '"client_email": "';
+    const emailFooter = '"client_id"';
+    const emailEndingToClip = '",';
+    const emailStart = privateKeyFile.indexOf(emailHeader) + emailHeader.length;
+    const emailEnd = privateKeyFile.indexOf(emailFooter);
+    const clientEmailNotClipped = privateKeyFile.substring(emailStart, emailEnd);
+    const emailClippedEnd = clientEmailNotClipped.indexOf(emailEndingToClip);
+    const clientEmail = clientEmailNotClipped.trim().substring(0, emailClippedEnd).replace("@", "%40");
```

Comment:
Instead of `.replace("@", "%40")`, use `encodeURIComponent`, which is what you meant.